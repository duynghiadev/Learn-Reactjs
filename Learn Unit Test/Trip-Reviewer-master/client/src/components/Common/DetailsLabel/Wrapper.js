import styled from 'styled-components';
import PropTypes from 'prop-types';

const Wrapper = styled.div`
  background: ${({ color }) => color};
  display: inline-block;
  padding: 5px 10px;
  border-radius: 3px;
  font-size: 12px;
  margin-right: 5px;
  color: #fff;
`;

Wrapper.propTypes = {
  color: PropTypes.string.isRequired,
};

export default Wrapper;
