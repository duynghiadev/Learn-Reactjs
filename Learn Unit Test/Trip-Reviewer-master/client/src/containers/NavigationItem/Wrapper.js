import styled from 'styled-components';
import PropTypes from 'prop-types';
import { Link } from 'react-router-dom';

const applySidenav = () => `
  color: #bababa;
  padding: 10px;
  border-top: 1px solid #404040;
  line-height: normal;
`;

const applyRegular = limit => `
  color: #ffffff;
  padding: 0 20px;
  display: inline-block;

  &:hover {
    background: #009eb3;
    transform: scale(1.05);
    text-shadow: 1px 1px 20px white;
  }

  &:active {
    background: #0495a8;
    transform: scale(1.01);
    text-shadow: 1px 1px 10px white;
  }

  @media (max-width: ${limit}px) {
    font-size: 16px;
    padding: 0 15px;
  }
`;

const Wrapper = styled(Link)`
  font-weight: 300;
  font-size: 18px;
  display: block;

  ${({ sidenav, limit }) => (sidenav ? applySidenav() : applyRegular(limit))}
`;

Wrapper.propTypes = {
  limit: PropTypes.number,
  sidenav: PropTypes.number,
};

Wrapper.defaultProps = {
  limit: 0,
  sidenav: 0,
};

export default Wrapper;
