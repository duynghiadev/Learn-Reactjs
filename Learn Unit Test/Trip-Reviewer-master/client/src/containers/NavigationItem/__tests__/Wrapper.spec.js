import React from 'react';
import { create } from 'react-test-renderer';

import Wrapper from '../Wrapper';

jest.mock('react-router-dom', () => ({ Link: 'Link' }));

const mockComponent = props => <Wrapper {...props} />;

describe('NavigationItem/<Wrapper />', () => {
  it('should render regular component', () => {
    const tree = create(mockComponent()).toJSON();

    expect(tree).toMatchSnapshot();
  });

  it('should render sidenav component', () => {
    const props = { sidenav: true };
    const tree = create(mockComponent(props)).toJSON();

    expect(tree).toMatchSnapshot();
  });
});
