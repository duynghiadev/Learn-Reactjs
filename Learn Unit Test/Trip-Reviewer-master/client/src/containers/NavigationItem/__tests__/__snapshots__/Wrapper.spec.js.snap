// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NavigationItem/<Wrapper /> should render regular component 1`] = `
<Link
  className="Wrapper-ThVjl iBnLkl"
/>
`;

exports[`NavigationItem/<Wrapper /> should render sidenav component 1`] = `
<Link
  className="Wrapper-ThVjl bwOZxy"
/>
`;
